Framework/Geometry/inc/MantidGeometry/Instrument/ComponentInfo.h

```diff
@@ -54,19 +54,24 @@ class MANTID_GEOMETRY_DLL ComponentInfo {
   /// Pointer to the actual ComponentInfo object (non-wrapping part).
   std::unique_ptr<Beamline::ComponentInfo> m_componentInfo;
   /// Collection of component ids
-  boost::shared_ptr<const std::vector<Mantid::Geometry::IComponent *>>
-      m_componentIds;
+  boost::shared_ptr<const std::vector<Geometry::IComponent *>> m_componentIds;
   /// Map of component ids to indexes
   boost::shared_ptr<const std::unordered_map<Geometry::IComponent *, size_t>>
       m_compIDToIndex;
 
+  /// Shapes for each component
+  boost::shared_ptr<std::vector<boost::shared_ptr<const Geometry::Object>>>
+      m_shapes;
```

Comment:
@SimonHeybrock Yes, I did think about storing the unique shapes separately and having raw pointers to those unique shapes. But I I don't know if the memory overhead, would really be worth the additional code + lifetime management issues. I think this could be an optimisation to consider when or if needed.